app/(dashboard)/dashboard/interviews/_actions/export.ts

```diff
@@ -32,81 +34,98 @@ type SuccessResult = {
 };
 
 export const exportSessions = async (
-  interviewIds: { id: string }[],
+  interviewIds: Interview['id'][],
   exportOptions?: ExportOptions,
 ) => {
-  const session = await getServerSession();
+  try {
+    const session = await getServerSession();
 
-  if (!session) {
-    throw new Error('You must be logged in to export interview sessions!.');
-  }
+    if (!session) {
+      throw new Error('You must be logged in to export interview sessions!.');
+    }
 
-  const installedProtocols = await api.protocol.get.all.query();
-  const interviewsSessions =
-    await api.interview.get.manyByIds.query(interviewIds);
+    // Get interviews From DB, by ids
+    const interviewsSessions =
+      await api.interview.get.forExport.query(interviewIds);
 
-  const fileExportManager = new FileExportManager(exportOptions);
+    // Store unique protocol ids in a Set
+    const installedProtocolIds = new Set<Protocol['id']>();
+    interviewsSessions.forEach((session) => {
+      installedProtocolIds.add(session.protocol.id);
+    });
 
-  fileExportManager.on('begin', () => {
-    // eslint-disable-next-line no-console
-    console.log({
-      statusText: 'Starting export...',
-      percentProgress: 0,
+    const installedProtocols: Protocol[] = [];
+
+    // find installed protocols based on the stored unique set of protocol ids
+    installedProtocolIds.forEach((id) => {
+      const foundSession = interviewsSessions.find(
+        (session) => session.protocol.id === id,
+      );
+      if (!foundSession) return;
+      installedProtocols.push(foundSession.protocol);
     });
-  });
 
-  fileExportManager.on('update', ({ statusText, progress }: UpdateItems) => {
-    // eslint-disable-next-line no-console
-    console.log({
-      statusText,
-      percentProgress: progress,
+    // The protocol object needs to be reformatted so that it is keyed by
+    // protocol.hash, since this is what network-exporters use.
+    const formattedProtocols = Object.values(installedProtocols).reduce(
+      (acc, protocol) => ({
+        ...acc,
+        [protocol.hash]: protocol,
+      }),
+      {},
+    );
+
```

Comment:
This whole section doesn't make any sense. You are doing multiple loops over the same data.

The whole point of the `Set` was that it would store `formattedProtocols`. The idea was to iterate over each session, check if the session's protocol was already in the set (using protocol.hash) and then if it wasn't, formatting the protocol and inserting it into the Set.

You're doing this process over three or four different steps here! 🤦 

If you want to use a Set, push an object representing the protocol. Alternatively, if you want to keep keying the protocols by their hash property, use a Map.

Reply:
Apologies for not simplifying this before. Done✅